presto-main/src/main/java/com/facebook/presto/operator/PagesIndex.java

```diff
@@ -343,7 +353,8 @@ public LookupSource createLookupSource(List<Integer> joinChannels, Optional<Inte
                 LookupSource lookupSource = lookupSourceFactory.createLookupSource(
                         valueAddresses,
                         ImmutableList.<List<Block>>copyOf(channels),
-                        hashChannel);
+                        hashChannel,
+                        createJoinFilterFunctionVerifier(filterFunction, ImmutableList.<List<Block>>copyOf(channels)));
```

Comment:
The `<List<Block>>` is not necessary in Java8.  You IDE should be warning you about this.